pandas/core/groupby.py

```diff
@@ -1297,11 +1297,18 @@ def apply(self, f, data, axis=0):
     @cache_readonly
     def indices(self):
         """ dict {group name -> group indices} """
+
+        def extract_values(x):
+            if isinstance(x, CategoricalIndex):
+                return x.values.get_values()
+            else:
+                return _values_from_object(x)
+
         if len(self.groupings) == 1:
             return self.groupings[0].indices
         else:
             label_list = [ping.labels for ping in self.groupings]
-            keys = [_values_from_object(ping.group_index) for ping in self.groupings]
+            keys = [extract_values(ping.group_index) for ping in self.groupings]
```

Comment:
instead of this, i think that ``.get_values()`` needs to be defined for ``CategoricalIndex`` (``.get_values`` is defined for a regular index and ``CategoricalIndex`` is just inhertiting).